test/0.8.9/withdrawal-queue-requests-finalization.test.js

```diff
@@ -71,6 +71,75 @@ contract('WithdrawalQueue', ([owner, daoAgent, user, anotherUser]) => {
     await snapshot.rollback()
   })
 
+  context('calculateFinalizationBatches', () => {
+    afterEach(async () => {
+      await snapshot.rollback()
+    })
+    it('reverts on invalid state', async () => {
+      await assert.reverts(
+        withdrawalQueue.calculateFinalizationBatches(shareRate(300), 100000, 1000, [
+          ETH(10),
+          true,
+          Array(36).fill(0),
+          0,
+        ]),
+        'InvalidState()'
+      )
+      await assert.reverts(
+        withdrawalQueue.calculateFinalizationBatches(shareRate(300), 100000, 1000, [0, false, Array(36).fill(0), 0]),
+        'InvalidState()'
+      )
+    })
+
+    it('works correctly on multiple calls', async () => {
+      const [requestId1, requestId2] = await withdrawalQueue.requestWithdrawals.call([ETH(1), ETH(1)], user, {
```

Comment:
can intercept the results via events or mocks (without dry-run calls, I mean)
or maybe it's better to introduce some kind of helper to call & run tx

Reply:
probably can just precalculate next two ids from queue length if there are no goal to test events